### PR TITLE
rubyPackages.ruby-libvirt: fix build with ruby 3.x

### PR DESCRIPTION
**Pull request by jnetod**
###### Motivation for this change

ZHF: #122042

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123445` at 14e182a5 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>rubyPackages.ruby-libvirt (rubyPackages_2_7.ruby-libvirt)</li>
    <li>rubyPackages_2_6.ruby-libvirt</li>
    <li>rubyPackages_3_0.ruby-libvirt</li>
    <li>tests.trivial</li>
  </ul>
</details>

File: pkgs/top-level/ruby-packages.nix [hunk 1]
```diff
@@ -2202,6 +2202,14 @@
       type = "gem";
     };
     version = "0.7.1";
+    dontBuild = false;
+    postPatch = ''
+      # https://gitlab.com/libvirt/libvirt-ruby/-/commit/43543991832c9623c00395092bcfb9e178243ba4
```
Comment (SuperSandro2000) [inline pkgs/top-level/ruby-packages.nix:2207]: Please fetchpatch the commit instead.

Comment (jnetod) [reply pkgs/top-level/ruby-packages.nix:2207]: There are commits prior to this commit that don't let it be applied cleanly. Should I cherry-pick it?

Beside this, I need to move these modifications to `pkgs/development/ruby-modules/gem-config/default.nix`

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 123445` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.trivial</li>
</ul>
</details>
<details>
<summary>
4 packages built:
</summary>
<ul>
<li>rubyPackages.ruby-libvirt (rubyPackages_2_7.ruby-libvirt)</li>
<li>rubyPackages_2_6.ruby-libvirt</li>
<li>rubyPackages_3_0.ruby-libvirt</li>
<li>vagrant</li>
</ul>
</details>

**Pull request merged**